src/components/scene/screenshot.js

```diff
@@ -51,36 +51,32 @@ module.exports.Component = registerComponent('screenshot', {
   },
 
   init: function () {
+    this.setupDone = false;
```

Comment:
don't need an extra flag. Can just check if there's a `self.canvas` in the setup function